asserts/ifacedecls.go

```diff
@@ -220,7 +222,263 @@ func compileAttributeConstraints(constraints interface{}) (*AttributeConstraints
 	return &AttributeConstraints{matcher: matcher}, nil
 }
 
+type fixedAttrMatcher struct {
+	result error
+}
+
+func (matcher fixedAttrMatcher) match(context string, v interface{}) error {
+	return matcher.result
+}
+
+var (
+	AlwaysMatchAttributes = &AttributeConstraints{matcher: fixedAttrMatcher{nil}}
+	NeverMatchAttributes  = &AttributeConstraints{matcher: fixedAttrMatcher{errors.New("not allowed")}}
+)
+
 // Check checks whether attrs don't match the constraints.
 func (c *AttributeConstraints) Check(attrs map[string]interface{}) error {
 	return c.matcher.match("", attrs)
 }
+
+func checkMapOrShortcut(context string, v interface{}) (m map[string]interface{}, outcomeShortcut string, err error) {
+	switch x := v.(type) {
+	case map[string]interface{}:
+		return x, "", nil
+	case string:
+		switch x {
+		case "true":
+			return nil, "true", nil
+		case "false":
+			return nil, "false", nil
+		}
+	}
+	return nil, "", fmt.Errorf("%s must be a map or one of the shortcuts 'true' or 'false'", context)
+
+}
+
+// PlugRule holds the rule of what is allowed, wrt installation and connection, for a plug of a specific interface for a snap.
```

Comment:
Minor nitpick-- I prefer when comments are wrapped to 80 characters (really, I prefer when everything is, but I realize that isn't the style here). This is especially true of github where you have to horizontal scroll.